smack-extensions/src/main/java/org/jivesoftware/smackx/message/correct/MessageCorrectExtension.java

```diff
@@ -0,0 +1,120 @@
+/**
+ *
+ * Copyright the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.message.correct;
+
+import java.io.IOException;
+
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.provider.ExtensionElementProvider;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ * An Extension that implements XEP-0308: Last Message Correction
+ * 
+ * This extension is expected to be added to message stanzas. Please refer to
+ * the XEP for more implementation guidelines.
+ * 
+ * @author Fernando Ramirez, f.e.ramirez94@gmail.com
+ */
+public class MessageCorrectExtension implements ExtensionElement {
+
+    /**
+     * The XML element name of a 'message correct' extension.
+     */
+    public static final String ELEMENT_NAME = "replace";
+
+    /**
+     * The namespace that qualifies the XML element of a 'message correct'
+     * extension.
+     */
+    public static final String NAMESPACE = "urn:xmpp:message-correct:0";
+
+    /**
+     * The id tag of a 'message correct' extension.
+     */
+    private static final String JID_TAG = "id";
+
+    /**
+     * The jid of the message to correct.
+     */
+    private String jidInitialMessage;
```

Comment:
That's no JID/XMPP address, this is just the message ID. Please rename this to `id`.